samwise.js

```diff
@@ -1,24 +1,25 @@
 /**
```

Comment:
The way you've refactored this complex method into simpler ones is great. It makes the code much more maintainable.